### PR TITLE
flashy: Support `grandcanyon` image meta validation

### PR DESCRIPTION
**Pull request by lhl2617**
# Summary
- Add `imageFormats` (a list of valid image partition formats) as arguments to `Validate` and `ValidateImageFile` so that validation can run against specific image formats.
- Add step for `grandcanyon` meta partition validation--corresponding diff in `fw-util`: https://github.com/facebook/openbmc/commit/bc32863c97e396e41a9c2054d2e2806562632310

## Discussion
For `grandcanyon` devices image partition validation essentially will be run _twice_, once via the common step, then once again in the newly added step. This might seem inefficient, but IMO is a necessary overhead if we were to stick to flashy's platform-based abstraction. There are potential workarounds:
  - Get the `buildname` within `flashy` itself, then obtain the right formats to validate against, as seen in #151 
    - breaks `flashy`'s platform-folder abstraction
  - Detect the presence of this step in the common validation step and skip validation there
    - Complex and again, breaks the abstraction
   
Thus, double-validation is done. Given that this extra step should not add much time to the whole flash process it should be a non-issue.

## Test plan
Unit tests `go test ./...`
testhard

### CODE REVIEW

**Comment by zhdaniel12** [comment]
Is all of these changes is for "GrandCanyon", blocking flashy from flashing non-vboot legacy image?

**Comment by lhl2617** [comment]
@zhdaniel12 yep :)

**Comment by zhdaniel12** [comment]
> @zhdaniel12 yep :)

Thanks submitting this. But the use case for retrofit from non-vboot image to vboot image is quite special, normally shall not happen in later NPI phases (i.e. PVT, Pilot or MP) when user start to use oobgrader to driven flashy do f/w upgrade/downgrade at scale automatically.

In early NPI phase, ad-hoc manual f/w upgrade is normally executed using fw-util or just flashcp etc raw utilities. 
This is why I sumbit fw-util: bc32863 changes to prevent user **manually** downgrading to early non-vboot image.
And change the image_parts.json file is the easiest way to implement.

So in short, we don't need implement this checking in flashy. Thanks again for submitting this PR, appreciated for your continuous support.

**Comment by lhl2617** [comment]
Thanks @zhdaniel12 ! I'll leave it here for reference in the event we need platform specific validation :)
